### PR TITLE
Initial SOMACollection class

### PR DESCRIPTION
**Pull request by johnkerl**
Context: #113

This adds a `SOMACollection` object, along with a simple wrapper script for command-line use.

More to come in terms of slicing operations etc as I pull more reviewable/mergeable pieces out of #80.

Also, more to come in separate PRs regarding examples/docs.

### CODE REVIEW

File: apis/python/src/tiledbsc/soma_collection.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    # ----------------------------------------------------------------
+    def add(self, soma: SOMA) -> None:
+        """
+        Adds a SOMA to the SOMACollection.
+        """
+        self._add_object(soma)
+
+    # ----------------------------------------------------------------
+    def remove(self, soma: SOMA) -> None:
```
[the remark this answers is not in the document]
Comment (johnkerl) [reply apis/python/src/tiledbsc/soma_collection.py:54]: agreed ... i can go either way on this ... let's iterate forward from here

File: apis/python/src/tiledbsc/soma_collection.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        """
+
+        if ctx is None and config is not None:
+            ctx = tiledb.Ctx(config)
+        if soma_options is None:
+            soma_options = SOMAOptions()  # Use default values from the constructor
+        super().__init__(
+            uri=uri,
+            name=name,
+            parent=parent,
```
[the remark this answers is not in the document]
Comment (johnkerl) [reply apis/python/src/tiledbsc/soma_collection.py:41]: https://github.com/single-cell-data/TileDB-SingleCell/blob/main/apis/python/src/tiledbsc/tiledb_object.py#L22-L41

[the remark this answers is not in the document]
Comment (aaronwolen) [reply apis/python/src/tiledbsc/soma_collection.py:41]: Ahh... missed that. Thanks for the reference.

[the remark this answers is not in the document]
Comment (johnkerl) [reply apis/python/src/tiledbsc/soma_collection.py:41]: @aaronwolen no worries, it's tucked away a few levels deep in the class-inheritance hierarchy :^/

**Pull request merged**